pheno_utils/questionnaires_handler.py

```diff
@@ -147,54 +147,77 @@ def transform_answers(
+    # Getting the data coding df from the large data coding csv
     code_df = mapping_df[mapping_df["code_number"] == code_string].copy()
     
-    #Make sure no leading 0s for coding values
+    # Make sure no leading 0s for coding values
     code_df["coding"] =  code_df["coding"].apply(convert_to_string)
     
     mapping_dict = dict(zip(code_df[code_from].astype(str), code_df[code_to]))
     
   
-  #adding fail safe incase older dictionaries don't have field type : TODO potentaily remove once older dictionaires are updated
-    if 'field_type' in dict_df.columns:
-        field_type =  dict_df.loc[tab_field_name]['field_type']
-        if isinstance(field_type, pd.Series):
-            if field_type.nunique() == 1:
-                field_type = field_type.iloc[0]
-            else:
-                warnings.warn(f"tabular field {tab_field_name} is used in 2 columns and have conflicting field types,please check and update dictionary. This field has not be converted.")
-                return orig_answer
-        
-        if field_type == 'Categorical (multiple)': 
-            normalise_answer = normalize_answers(orig_answer, field_type)
-            check_invalid_values(normalise_answer , code_df)
-            transformed_answer = normalise_answer.apply(replace_values, mapping_dict = mapping_dict)
+    field_type =  dict_df.loc[tab_field_name]['field_type']
+    if isinstance(field_type, pd.Series):
+        if field_type.nunique() == 1:
+            field_type = field_type.iloc[0]
         else:
-            #if categorical single
-            normalized_answer = normalize_answers(orig_answer, field_type)
-            check_invalid_values(normalized_answer, code_df)
-            transformed_answer = normalized_answer.replace(mapping_dict)
-            transformed_answer = transformed_answer.astype("category")
-
-        return transformed_answer
+            warnings.warn(f"tabular field {tab_field_name} is used in 2 columns and have conflicting field types,please check and update dictionary. This field has not be converted.")
+            return orig_answer
+    
+    if field_type == 'Categorical (multiple)': 
+        normalise_answer = normalize_answers(orig_answer, field_type)
+        check_invalid_values(normalise_answer , code_df)
+        transformed_answer = normalise_answer.apply(replace_values, mapping_dict = mapping_dict)
     else:
-        return orig_answer
-
+        normalized_answer = normalize_answers(orig_answer, field_type)
+        check_invalid_values(normalized_answer, code_df)
+        transformed_answer = normalized_answer.replace(mapping_dict)
+        transformed_answer = transformed_answer.astype("category")
 
+    return transformed_answer
+    
+def convert_codings_to_int(df: pd.DataFrame, dict_df: pd.DataFrame, fields_for_translation: list, preferred_language: str) -> pd.DataFrame:
+    df_coding = df.copy()
+    if preferred_language == 'coding':
+        for column in fields_for_translation:
```

Comment:
since this loops mimicks the one in the calling function, I'm a little tempted to have this function handle a single column, and call it in the loop of transform_dataframe. but not sure it's worth your time to refactor.